src/delta.rs

```diff
@@ -114,12 +114,27 @@ where
                 handle_hunk_meta_line(&mut painter, &line, config)?;
                 continue;
             }
-        } else if source == Source::DiffUnified && line.starts_with("Only in ") {
+        } else if source == Source::DiffUnified && line.starts_with("Only in ")
+            || line.starts_with("Submodule ")
+        {
+            // Additional FileMeta cases:
+            //
+            // 1. When comparing directories with diff -u, if filenames match between the
+            //    directories, the files themselves will be compared. However, if an equivalent
+            //    filename is not present, diff outputs a single line (Only in...) starting
+            //    indicating that the file is present in only one of the directories.
+            //
+            // 2. Git diff emits lines describing submodule state such as "Submodule x/y/z contains
+            //    untracked content"
+            //
+            // See https://github.com/dandavison/delta/issues/60#issuecomment-557485242 for a
+            // proposal for more robust parsing logic.
+
             state = State::FileMeta;
             painter.paint_buffered_lines();
             if config.opt.file_style != cli::SectionStyle::Plain {
                 painter.emit()?;
-                handle_directory_diff_unique_file_name(&mut painter, &raw_line, config)?;
+                handle_generic_file_meta_header_line(&mut painter, &raw_line, config)?;
```

Comment:
@m-lima I thought that your function `handle_directory_diff_unique_file_name` could actually be used more broadly to implement all `FileMeta` output, so I repurposed it and moved your explanatory comments to this call site.

Reply:
Nice! Makes a lot of sense.
Especially if you decide in the future change the appearance of some of the `FileMeta` lines, you can easily plug them in 👍

Reply:
Nice! Makes a lot of sense.
Especially if you decide in the future change the appearance of some of the `FileMeta` lines, you can easily plug them in 👍